--- a/freeplane/src/org/freeplane/main/application/FreeplaneStarter.java
+++ b/freeplane/src/org/freeplane/main/application/FreeplaneStarter.java
@@ -158,7 +158,7 @@
 			LinkController.install();
 			IconController.install();
 			HelpController.install();
-			controller.addAction(new UpdateCheckAction());
+			//controller.addAction(new UpdateCheckAction());
 			controller.addAction(new NextNodeAction(Direction.FORWARD));
 			controller.addAction(new NextNodeAction(Direction.BACK));
 			controller.addAction(new NextNodeAction(Direction.FORWARD_N_FOLD));
